inference_rules.adoc

```diff
@@ -821,6 +821,30 @@ The DLRMv2 MLPerf inference code has an option to aggregate multiple consecutive
 2. drawn uniformly from interval [`--samples-to-aggregate-min`, `--samples-to-aggregate-max`]	
 3. drawn from a custom distribution, with its quantile (inverse of CDP) specified in `--samples-to-aggregate-quantile-file=./tools/dist_quantile.txt`.
 
+=== LLM Benchmarks
+
+
+LLM benchmarks contain an auto-regressive generation loop. This loop’s execution is controlled by a decoding algorithm. The following rules discuss the decoding algorithm in the light of MLPerf Inference rules for the Closed submissions:
+
+* Decoding algorithm should follow the implementation available on huggingface: https://huggingface.co/blog/how-to-generate#beam-search 
```

Comment:
Please explain beam search algorithm in this document.

[https://huggingface.co/blog/how-to-generate#beam-search ](https://huggingface.co/blog/how-to-generate#beam-search) does not contain any implementation. Please describe how beam search works on a toy example and do not refer to HF docs. We believe that in the inference_rules the algorithm itself should be described, not its implementation.

In HF's beam_search algorithm we can find that topK with k=num_beams*2 is computed: [https://github.com/huggingface/transformers/blob/v4.30.0/src/transformers/generation/utils.py#L2955](https://github.com/huggingface/transformers/blob/v4.30.0/src/transformers/generation/utils.py#L2955). The document "How to generate beam search?" does not mention it and simplifies the algorithm a lot comparing with the actual implementation of beam_search in transformers library. How much freedom does submitter have?